generators/cloud-init.go

```diff
@@ -117,18 +120,28 @@ local-hostname: {{ container.name }}
 {{ config_get("user.meta-data", "") }}
 `
 	case "vendor-data":
-		content = `{{ config_get("user.vendor-data", properties.default) }}
+		content = `{%- if config_get("cloud-init.user-data", properties.default) == properties.default -%}
+{{ config_get("user.user-data", properties.default) }}
+{%- else -%}
+{{- config_get("cloud-init.user-data", properties.default) }}
+{%- endif -%}
```

Comment:
All of this should be `vendor-data`, not `user-data`, no?

Reply:
Fixed